rest-service/manager_rest/rest/rest_utils.py

```diff
@@ -719,6 +719,39 @@ def decrease_deployment_counts_in_graph(self, target_id, source):
                         target.sub_environments_count -= total_environments
                     self.sm.update(target)
 
+    def update_deployment_counts_after_source_conversion(self,
+                                                         source,
+                                                         convert_type):
+        """
+        Update counter for deployment that is already connected to other
+        deployments where deployment could be changed from `service` to
+        `environment` and vice versa and we need to take care of deployment
+        counters so that count will not mess up
+        :param source: Deployment source instance that its type get changed
+        :param convert_type: The type we need to convert source deployment
+        to. It could be changed from `service` to `environment` and vice versa
+        """
+        if convert_type not in ('environment', 'service',):
+            return
```

Comment:
Throw an error?

Reply:
well, I do not want to throw error here because user could add values other than these two which requires no need to update the counter

Reply:
Unless I missed something, this would never get here with `None`.

Reply:
Yeah, it will not and it could be removed